src/SharpCompress/Compressor/ADC/ADCBase.cs

```diff
@@ -0,0 +1,210 @@
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+using System;
+using System.IO;
+
+namespace SharpCompress.Compressor.ADC
+{
+    /// <summary>
+    /// Provides static methods for decompressing Apple Data Compression data
+    /// </summary>
+    public static class ADCBase
+    {
+        const int Plain = 1;
+        const int TwoByte = 2;
+        const int ThreeByte = 3;
+
+        static int GetChunkType(byte byt)
+        {
+            if ((byt & 0x80) == 0x80)
+                return Plain;
+            if ((byt & 0x40) == 0x40)
+                return ThreeByte;
+            return TwoByte;
+        }
+
+        static int GetChunkSize(byte byt)
+        {
+            switch (GetChunkType(byt))
+            {
+                case Plain:
+                    return (byt & 0x7F) + 1;
+                case TwoByte:
+                    return ((byt & 0x3F) >> 2) + 3;
+                case ThreeByte:
+                    return (byt & 0x3F) + 4;
+                default:
+                    return -1;
+            }
+        }
+
+        static int GetOffset(byte[] chunk, int position)
+        {
+            switch (GetChunkType(chunk[position]))
+            {
+                case Plain:
+                    return 0;
+                case TwoByte:
+                    return ((chunk[position] & 0x03) << 8) + chunk[position + 1];
+                case ThreeByte:
+                    return (chunk[position + 1] << 8) + chunk[position + 2];
+                default:
+                    return -1;
+            }
+        }
+
+        /// <summary>
+        /// Decompresses a byte buffer that's compressed with ADC
+        /// </summary>
+        /// <param name="input">Compressed buffer</param>
+        /// <param name="output">Buffer to hold decompressed data</param>
+        /// <param name="bufferSize">Max size for decompressed data</param>
+        /// <returns>How many bytes are stored on <paramref name="output"/></returns>
+        public static int Decompress(byte[] input, out byte[] output, int bufferSize = 262144)
+        {
+            return Decompress(new MemoryStream(input), out output, bufferSize);
+        }
+
+        /// <summary>
+        /// Decompresses a stream that's compressed with ADC
+        /// </summary>
+        /// <param name="input">Stream containing compressed data</param>
+        /// <param name="output">Buffer to hold decompressed data</param>
+        /// <param name="bufferSize">Max size for decompressed data</param>
+        /// <returns>How many bytes are stored on <paramref name="output"/></returns>
+        public static int Decompress(Stream input, out byte[] output, int bufferSize = 262144)
```

Comment:
Rather than this method signature, couldn't this be the same as in `Read(byte[], int, int)`?

Why buffer at all?  Let the Stream contract control the buffer size.  There's a `BufferedStream` class that handles buffering if performance is improved by having a larger buffer size.